src/status_im2/common/nickname_drawer/view.cljs

```diff
@@ -0,0 +1,83 @@
+(ns status-im2.common.nickname-drawer.view
+  (:require [clojure.string :as string]
+            [quo2.foundations.colors :as colors]
+            [quo2.components.icon :as icons]
+            [quo2.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.common.nickname-drawer.style :as style]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn empty-nickname?
+  [nickname]
+  (string/blank? nickname))
+
+(defn add-nickname-toast
+  [primary-name entered-nickname public-key]
+  (rf/dispatch [:hide-bottom-sheet])
+  (rf/dispatch [:toasts/upsert
+                {:id         :add-nickname
+                 :icon       :i/correct
+                 :icon-color (colors/theme-colors colors/success-60 colors/success-50)
+                 :text       (i18n/label
+                              :t/set-nickname-toast
+                              {:primary-name primary-name
+                               :nickname     (string/trim entered-nickname)})}])
+  (rf/dispatch [:contacts/update-nickname public-key (string/trim entered-nickname)]))
+
+(defn nickname-drawer
+  [{:keys [title description contact accessibility-label
+           close-button-text]}]
+  (let [{:keys [primary-name nickname public-key]} contact
+        entered-nickname                           (reagent/atom (or nickname ""))
+        photo-path                                 (when-not (empty? (:images contact))
+                                                     (rf/sub [:chats/photo-path public-key]))]
+    (fn []
```

Comment:
I told people to fix it and not I am doing it myself :)

Reply:
It's hard to remember this, and I forgot many times to repeat args. It's also fragile, since devs need to always remember to review and maybe update both function signatures (outer & inner). It's not just a rookie mistake, IMO it's an issue with how the Reagent API works, which allows for such hard to spot bugs ;)